conda_build/metadata.py

```diff
@@ -628,41 +632,40 @@ def finalize_outputs_pass(base_metadata, render_order, pass_no, outputs=None,
             if base_metadata.config.verbose:
                 log.info("Attempting to finalize metadata for {}".format(metadata.name()))
             om = base_metadata.copy()
+            om.other_outputs = metadata.other_outputs
```

Comment:
other_outputs is the context of what's available for pin_subpackage.  It's stored on the ``metadata`` object here, but not on ``base_metadata``, which ``om`` is a copy of.  Before we do re-rendering of om's metadata, we need to have other_outputs in place, so it can refer to it for any pin_subpackage stuff it has.